scipy/_distributor_init.py

```diff
@@ -0,0 +1,35 @@
+""" Distributor init file
+
+Distributors: you can add custom code here to support particular distributions
+of numpy.
+
+For example, this is a good place to put any checks for hardware requirements.
+
+The numpy standard source distribution will not put code in this file, so you
+can safely replace this file with your own version.
+"""
+
+import ctypes
+import os
+import sys
+
+# Patch the DLL search path on windows so that extensions can load any DLLs that
+# they depend on (specifically gfortran DLLs) on windows.
+
+if sys.platform == 'win32':
+    try:
+        AddDllDirectory = ctypes.windll.kernel32.AddDllDirectory
+        AddDllDirectory.argtypes = [ctypes.c_wchar_p]
+        # Needed to initialize AddDllDirectory modifications
+        ctypes.windll.kernel32.SetDefaultDllDirectories(0x1000)
+    except AttributeError:
+        AddDllDirectory = ctypes.windll.kernel32.SetDllDirectoryW
+        AddDllDirectory.argtypes = [ctypes.c_wchar_p]
+
+    def patch_dll_search_path():
```

Comment:
You've covered the case where AddDllDirectory is missing (some copies of Windows 7 apparently), but I'm worried about the `SetDllDirectoryW` call.  See:

* https://github.com/pypa/wheel-builders/pull/2/files#diff-37a1ef08dd1b0096d769190274121258R378
* https://github.com/numpy/numpy/wiki/windows-dll-notes

In particular, I believe I'm right in saying that `SetDllDirectory` will override any previous call to `SetDllDirectory`, so using it has the potential to make other things in this process stop working.  I'm sure that's rare, but still, I wonder whether it would be better to modify PATH instead.